contracts/account/library.cairo

```diff
@@ -0,0 +1,558 @@
+%lang starknet
+
+from starkware.cairo.common.cairo_builtins import HashBuiltin, SignatureBuiltin
+from starkware.cairo.common.signature import verify_ecdsa_signature
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.memcpy import memcpy
+from starkware.cairo.common.math import assert_not_zero, assert_le, assert_nn
+from starkware.starknet.common.syscalls import (
+    library_call,
+    call_contract,
+    get_tx_info,
+    get_contract_address,
+    get_caller_address,
+    get_block_timestamp,
+)
+from starkware.cairo.common.bool import TRUE, FALSE
+
+from contracts.upgrade.Upgradable import _set_implementation
+
+const SUPPORTS_INTERFACE_SELECTOR = 1184015894760294494673613438913361435336722154500302038630992932234692784845;
+const ERC165_ACCOUNT_INTERFACE = 0xf10dbd44;
+
+/////////////////////
+// STRUCTS
+/////////////////////
+
+struct Call {
+    to: felt,
+    selector: felt,
+    calldata_len: felt,
+    calldata: felt*,
+}
+
+// Tmp struct introduced while we wait for Cairo
+// to support passing `[Call]` to __execute__
+struct CallArray {
+    to: felt,
+    selector: felt,
+    data_offset: felt,
+    data_len: felt,
+}
+
+struct Escape {
+    active_at: felt,
+    type: felt,
+}
+
+/////////////////////
+// EVENTS
+/////////////////////
+
+@event
+func signer_changed(new_signer: felt) {
+}
+
+@event
+func guardian_changed(new_guardian: felt) {
+}
+
+@event
+func guardian_backup_changed(new_guardian: felt) {
+}
+
+@event
+func escape_guardian_triggered(active_at: felt) {
+}
+
+@event
+func escape_signer_triggered(active_at: felt) {
+}
+
+@event
+func escape_canceled() {
+}
+
+@event
+func guardian_escaped(new_guardian: felt) {
+}
+
+@event
+func signer_escaped(new_signer: felt) {
+}
+
+@event
+func account_upgraded(new_implementation: felt) {
```

Comment:
any reason why `account_created` takes the account address but `account_upgraded` does not?

Reply:
Not really, no. The event is associated to the contract emitting so there is no need to pass the account address in the event. However, `account_created` is defined in a discussion outside Argent and I don't want to change it before this discussion moves to a standard.